test/quantization/pt2e/test_quantize_pt2e.py

```diff
@@ -811,6 +811,152 @@ def validate(self, model: torch.fx.GraphModule) -> None:
             m, expected_node_list=node_list, expected_node_occurrence=node_occurrence
         )
 
+    def test_shared_qspec_transitivity(self):
+        """This tests the transitivity of SharedQuantizationSpec, that is
+        if A is shared with B, B is shared with C, then C should be shared with A as well
+        """
+        class BackendAQuantizer(Quantizer):
```

Comment:
we should factor this quantizer out in common util. There is a lot of code duplication here

Reply:
OK, I'll add a TODO